### PR TITLE
[TCSACR-595] Add new guide documents for Tizen.Core

### PR DESCRIPTION
**Pull request by hjhun**
### Change Description ###

This PR adds new guide documents for Tizen.Core API.

### API Changes ###

In case of ACR, put the link of ACR

For instance,

 -TCSACR-595

### CODE REVIEW

**Review by safir-srbd — COMMENTED** [review]
@hjhun Please also add the files to the `toc_all.md` file, so they appear in the Left Navigation Bar (LNB).

**Comment by hjhun** [comment]
> @hjhun Please also add the files to the `toc_all.md` file, so they appear in the Left Navigation Bar (LNB).

Done

File: docs/application/dotnet/guides/applications/tizen-core/tizen-core-channel.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+task.Send(async () => {
+    var channelObject = await channel.Receiver.Receive();
+    Log.Debug(LogTag, "Received Message = " + (string)channelObject.Data);
+    channelObject.Dispose();
+});
+
+The `await channel.Receiver.Receive()` blocks indefinitely if there is no data being transmitted. It is recommended to use it in situations where the sender sends data reliably.
+After receiving the object, its ownership is transferred to the receiving party, so you must release it by calling the `Tizen.Core.ChannelObject.Dispose()`.
+
+## Related Information
```
Comment (safir-srbd) [inline docs/application/dotnet/guides/applications/tizen-core/tizen-core-channel.md:83]:
```suggestion
## Related information
```

File: docs/application/dotnet/guides/applications/tizen-core/tizen-core-event.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+string message = "Event Message";
+using (var eventObject = new EventObject(id++, message))
+{
+    task.EmitEvent(eventObject);
+}
+
+```
+If you want to deliver `Tizen.Core.EventObject` only to a specific `Tizen.Core.Event`, use `Tizen.Core.Event.Emit()` method instead.
+
+## Related Information
```
Comment (safir-srbd) [inline docs/application/dotnet/guides/applications/tizen-core/tizen-core-event.md:58]:
```suggestion
## Related information
```

File: docs/application/dotnet/guides/applications/tizen-core/tizen-core.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+Here's an example of registering a timer that calls the callback every 100 ms.
+```csharp
+var task = TizenCore.FindFromCurrentThread();
+var timerId = task.AddTimer(100, () => {
+    Log.Debug(LogTag, "Timer is invoked");
+});
+```
+If the registered timer source is no longer needed, you should remove it by calling `Tizen.Core.Task.RemoveTimer()` with the `timerId`.
+
+## Related Information
```
Comment (safir-srbd) [inline docs/application/dotnet/guides/applications/tizen-core/tizen-core.md:112]:
```suggestion
## Related information
```

**Review by safir-srbd — COMMENTED** [review]
@hjhun Please check and implement the suggested changes.

File: docs/application/dotnet/guides/applications/tizen-core/tizen-core.md (hunk 3 above)
Comment (hjhun) [reply docs/application/dotnet/guides/applications/tizen-core/tizen-core.md]: Done

File: docs/application/dotnet/guides/applications/tizen-core/tizen-core-event.md (hunk 2 above)
Comment (hjhun) [reply docs/application/dotnet/guides/applications/tizen-core/tizen-core-event.md]: Done

File: docs/application/dotnet/guides/applications/tizen-core/tizen-core-channel.md (hunk 1 above)
Comment (hjhun) [reply docs/application/dotnet/guides/applications/tizen-core/tizen-core-channel.md]: Done

**Review by safir-srbd — COMMENTED** [review]
@hjhun LGTM.

**Review by safir-srbd — APPROVED** [review]

**Pull request merged**